confidant/models/credential.py

```diff
@@ -142,6 +160,40 @@ def _get_decrypted_credential_pairs(self):
         _credential_pairs = json.loads(_credential_pairs)
         return _credential_pairs
 
+    @property
+    def next_rotation_date(self):
+        """
+        Return when a credential needs to be rotated for security purposes.
+        """
+        if not self.requires_rotation or self.rotation_frequency is None:
+            return None
+
+        # If a credential has never been rotated or been decrypted,
+        # immediately rotate
+        if self.last_rotation_date is None:
+            return datetime.utcnow()
+
+        if self.last_decrypted_date and \
+                self.last_decrypted_date > self.last_rotation_date:
+            return datetime.utcnow()
+
+        return self.last_rotation_date + timedelta(days=self.rotation_frequency)
+
+    @property
+    def rotation_frequency(self):
+        """
+        Return how frequently a credential needs to be rotated
+        Return None if category is not in CREDENTIAL_ROTATION_DAYS config
+        """
+        return settings.CREDENTIAL_ROTATION_DAYS.get(self.category)
+
+    @property
+    def requires_rotation(self):
+        """
+        Some credentials need to be periodically rotated
+        """
+        return self.category == settings.FINANCIALLY_SENSITIVE
```

Comment:
For this, you don't want to go off a specific tag, you want to go off `self.next_rotation_date`